lib/pysquared/logger.py

```diff
@@ -6,12 +6,31 @@
 import json
 import time
 
+from lib.pysquared.nvm.counter import Counter
+
+
+class LogLevel:
+    NOTSET = 0
+    DEBUG = 1
+    INFO = 2
+    WARNING = 3
+    ERROR = 4
+    CRITICAL = 5
+
 
 class Logger:
-    def __init__(self) -> None:
-        self.logToStandardOut: bool = True
+    def __init__(
+        self,
+        error_counter: Counter,
+        log_level: int = LogLevel.NOTSET,
+    ) -> None:
+        self._error_counter: Counter = error_counter
+        self._log_level: int = log_level
 
-    def _log(self, level: str, message: str, **kwargs) -> None:
+    def can_print_this_level(self, level_value: int) -> bool:
```

Comment:
This is also probably not intended to be called outside of this class. Let's make it internal.
```suggestion
    def _can_print_this_level(self, level_value: int) -> bool:
```